### DISCUSSION TITLE
helm-follow-mode keeps turning off

### DISCUSSION THREAD

**Discussion by rexackermann** [emacs-helm/helm#2602] [Q&A] [answered]
I am using doom emacs.I want to preview the files when I do helm-find-files.So, I used C-c C-f to toogle on helm-follow-mode.It works at first.But when i go over a folder it just turns of and I have to manually turn it on again.That's a hastle.Can't I just keep it on all the time ? How do I do it ?

**Comment by thierryvolpiatto** [comment]
Rex Ackermann ***@***.***> writes:

> But when i go over a folder it just turns of

This to prevent having dozen of dired buffers open though you can kill
the buffer without quitting Helm.

> and I have to manually turn it on again.

You can use C-<down> to skip the folder (the behavior is the contrary
when follow-mode is disabled).

> Can't I just keep it on all the time ?

No, at least for now.

-- 
Thierry

**Reply by rexackermann** [reply]
Thanks for the reply.I understand it's a feature,Not a bug.
But ,can I still keep it on though.Make it turn on automatically when I am not on a folder or the things that usually are blocked ? Using some hook maybe ? Which will reactivate it when I pass the file and a supported one is selected ?

I actually ,recently started using doom emacs. I used neovim with telescope plugin before which had great support for file preview.I found this project very similar except this(You know the continuous file preview part) doesn't work the way I wanted.It would be very helpful if you could help me out with this or point me the right way.

**Comment by thierryvolpiatto** [comment]
Rex Ackermann ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
> Thanks for the reply.I understand it's a feature,Not a bug.  But ,can
> I still keep it on though.Make it turn on automatically when I am not
> on a folder or the things that usually are blocked ? Using some hook
> maybe ? Which will reactivate it when I pass the file and a supported
> one is selected ?
>
> I actually ,recently started using doom emacs. I used neovim with
> telescope plugin before which had great support for file preview.I
> found this project very similar except this(You know the continuous
> file preview part) doesn't work the way I wanted.It would be very
> helpful if you could help me out with this or point me the right way.

I can add a variable (something like
helm-ff-allow-following-on-directory), is it ok for you to have a dired
buffer popping up in such case?

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>


-- 
Thierry

**Comment by thierryvolpiatto** [comment, marked as answer]
Done, the variable is `helm-ff-ignore-following-on-directory`, the directories are just ignored, no dired buffer popping up.

**Comment by rexackermann** [comment]
Thanks ,but I can't make it work.I added this line,
```
(setq helm-ff-ignore-following-on-directoriy t)
```

Am I following right ? It should now keep the helm-follow-mode on when I'm passing by a directory, right ?
Did I do it wrong ?

Thanks again.

**Comment by thierryvolpiatto** [comment]
Rex Ackermann ***@***.***> writes:

> Thanks ,but I can't make it work.I added this line,
>
> (setq helm-ff-ignore-following-on-directoriy t)
                                           ^^^
> Am I following right ? It should now keep the helm-follow-mode on when I'm passing by a directory, right ?
> Did I do it wrong ?

Yes, typo.

-- 
Thierry

**Comment by rexackermann** [comment]
Is this the right one ?
```
(setq helm-ff-ignore-following-on-directory t)
```
Okay ,normally the variable is highlighted yollow.But this one stays white.If it helps.

**Comment by thierryvolpiatto** [comment]
Rex Ackermann ***@***.***> writes:

> Is this the right one ?

Yes it is the right one.

> (setq helm-ff-ignore-following-on-directory t)

Ok, that's now correct.
Tip: Always use some kind of completion when writing symbols, this avoid
typo errors.

> Okay ,normally the variable is highlighted yollow.But this one stays white.If it helps.

No, when setq'ing a variable it is not highlighted, only its definition
in source file is highlighted with font-lock-variable-name-face.

-- 
Thierry

**Reply by rexackermann** [reply]
> Ok, that's now correct. Tip: Always use some kind of completion when writing symbols, this avoid typo errors.

It isn't shwowing up in the completion.

Thanks for dealing with my annoying issues.I am getting a feeling that it might be related to doom emacs. Maybe I will try to figure it out.But for now I guess I will have to stick with C-up/down.It's nice to know that the functionality exits now regardless.I will mark your 3rd comment as the answer.

Thank u very much again.

**Comment by thierryvolpiatto** [comment]
Rex Ackermann ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
>     Ok, that's now correct. Tip: Always use some kind of completion when writing symbols, this avoid typo errors.
>     It isn't shwowing up in the completion.

Did you have upgraded helm?
    
> Thanks for dealing with my annoying issues.I am getting a feeling that
> it might be related to doom emacs.

If you don't have the variable, this mean you haven't installed the latest
Helm, please upgrade.

> Maybe I will try to figure it out.But for now I guess I will have to
> stick with C-up/down.It's nice to know that the functionality exits
> now regardless.I will mark your 3rd comment as the answer.
>
> Thank u very much again.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>


-- 
Thierry

**Reply by rexackermann** [reply]
I tried doom upgrade.The results are same.It's okay.Thanks for trying.
Well if you want to investigate for other reasons
The config is here : [config](https://github.com/rexackermann/rexdoom)
